src/components/workspace/window-contents/window-content-factory.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { Box } from '@mui/material';
+import { ReactFlowProvider } from '@xyflow/react';
+import { memo } from 'react';
+import NetworkModificationTreePane from '../../network-modification-tree-pane';
+import { SpreadsheetView } from '../../spreadsheet-view/spreadsheet-view';
+import { ReportViewerTab } from '../../report-viewer-tab.jsx';
+import { ResultViewTab } from '../../result-view-tab';
+import ParametersTabs from '../../parameters-tabs';
+import { isNodeBuilt } from '../../graph/util/model-functions';
+import { WindowType, type DiagramWindowData } from '../types/workspace.types';
+import type { CurrentTreeNode } from '../../graph/tree-node.type';
+import type { UUID } from 'node:crypto';
+import { DiagramWindowContent } from './diagrams/diagram-window-content';
+import { MapWindowContent } from './map-window-content';
+import NodeEditor from 'components/graph/menus/network-modifications/node-editor';
+
+export interface WindowContentDependencies {
+    studyUuid: string | null;
+    currentRootNetworkUuid: string | null;
+    currentNode: CurrentTreeNode | null;
+}
+
+export const WindowContentFactory = memo(
+    ({
+        windowType,
+        windowData,
+        windowId,
+        dependencies,
+    }: {
+        windowType: WindowType;
+        windowData: unknown;
+        windowId: string;
+        dependencies: WindowContentDependencies;
+    }): React.ReactNode => {
+        const { studyUuid, currentRootNetworkUuid, currentNode } = dependencies;
+
+        if (!studyUuid || !currentRootNetworkUuid || !currentNode) {
+            return <Box>Invalid window dependencies</Box>;
+        }
+
+        switch (windowType) {
+            case WindowType.TREE:
+                return (
+                    <ReactFlowProvider>
+                        <NetworkModificationTreePane
+                            studyUuid={studyUuid}
+                            currentRootNetworkUuid={currentRootNetworkUuid}
```

Comment:
we could simplify
all those parameters studyUuid, currentRootNetworkUuid, currentNode, isNodeBuilt could all be retrieved from redux by the child component itself

from the types and design of the workspace, only DIAGRAM  and SPREADSHEET should need parameters, the data you've specified

If window.data depended on window.type it would also help